File: algorithm-case/sort/insertion.js
const insertionSort = (arrayList, asc = true) => {
  for (let i = 1; i < arrayList.length; i++) {
    let currentValue = arrayList[i];
    let checkedIndex = i - 1;
    while (checkedIndex >= 0) {
      if (currentValue < arrayList[checkedIndex] && asc) {
        arrayList[checkedIndex + 1] = arrayList[checkedIndex];
        arrayList[checkedIndex] = currentValue;
      } else if (currentValue > arrayList[checkedIndex] && asc) {
        break;
      }

      if (currentValue > arrayList[checkedIndex] && !asc) {
        arrayList[checkedIndex + 1] = arrayList[checkedIndex];
        arrayList[checkedIndex] = currentValue;
      } else if (currentValue < arrayList[checkedIndex] && !asc) {
        break;
      }
      checkedIndex--;
    }
  }
  return arrayList;
};

const assertAsc = [1, 2, 3, 4, 5, 5];
// const assertDesc = [5, 5, 4, 3, 2, 1];
const random = [3, 5, 2, 1, 5, 4];
const resultAsc = insertionSort([].concat(random));
// const resultDes = insertionSort(random, false);
console.log(`randomElement: ${random}`);
console.log(`resultElement: ${resultAsc}`);
console.log(`Sorting result: ${assertAsc.toString() === resultAsc.toString()}`);

File: algorithm-case/sort/insertion.py
def insertionSort(arrayList, asc=True):
    for i in range(1, len(arrayList)):
        currentValue = arrayList[i]
        checkedIndex = i - 1
        while(checkedIndex >= 0):
            if (currentValue < arrayList[checkedIndex] and asc):
                arrayList[checkedIndex + 1] = arrayList[checkedIndex]
                arrayList[checkedIndex] = currentValue
            elif (currentValue > arrayList[checkedIndex] and asc):
                break

            if (currentValue > arrayList[checkedIndex] and not asc):
                arrayList[checkedIndex + 1] = arrayList[checkedIndex]
                arrayList[checkedIndex] = currentValue
            elif (currentValue < arrayList[checkedIndex] and not asc):
                break
            checkedIndex -= 1
    return arrayList


assertAsc = [1, 2, 3, 4, 5, 5]
# assertDes = [5, 5, 4, 3, 2, 1]
random = [3, 5, 2, 1, 5, 4]
resultAsc = insertionSort(random.copy())
# resultDes = bubbleSort(random.copy(), False)
print('randomElement: ', random)
print('resultElement: ', resultAsc)
print('Sorting result: ', assertAsc == resultAsc)
